tools/export/sw4stm32/__init__.py

```diff
@@ -14,95 +14,374 @@
+        {
+            'name': 'NUCLEO-L476RG',
+            'mcuId': 'STM32L476RGTx'
+        },
     }
 
     TARGETS = BOARDS.keys()
 
-    def __gen_dir(self, dirname):
-        settings = join(self.export_dir, dirname)
+    def __gen_dir(self, dir_name):
+        """
+        Method that creates directory
+        """
+        settings = join(self.export_dir, dir_name)
         mkdir(settings)
 
     def __generate_uid(self):
+        """
+        Method to create random int
+        """
         return "%0.9u" % randint(0, 999999999)
 
+    @staticmethod
+    def filter_dot(path):
+        """
+        This function removes ./ from str.
+        str must be converted with win_to_unix() before using this function.
+        """
+        if not path:
+            return path
+        if path.startswith('./'):
+            return path[2:]
+        return path
+
+    def build_excludelist(self):
+        """
+        This method creates list for excluded directories.
+        """
+        self.source_folders = [self.filter_dot(s)
+                               for s in set(dirname(src) for src in
+                                            self.resources.c_sources +
+                                            self.resources.cpp_sources +
+                                            self.resources.s_sources)]
+        if '.' in self.source_folders:
+            self.source_folders.remove('.')
+
+        top_folders = [f for f in set(s.split('/')[0]
+                                      for s in self.source_folders)]
+
+        for top_folder in top_folders:
+            for root, dirs, files in walk(top_folder, topdown=True):
+                # Paths returned by os.walk() must be split with os.dep
+                # to accomodate Windows weirdness.
+                parts = root.split(sep)
+                self.remove_unused('/'.join(parts))
+
+    def remove_unused(self, path):
+        """
+        Method for checking if path is needed.
+        Method adds path to excluded list if not needed
+        and is not subdirectory of already excluded directory
+        """
+        found = path in self.include_path
+        needtoadd = True
+        if not found:
+            for directory in self.exclude_dirs:
+                # Do not exclude subfolders from excluded folder
+                if directory + '/' in path:
+                    needtoadd = False
+            if needtoadd:
+                self.exclude_dirs.append(path)
+
     def generate(self):
+        """
+        Generate the .project and .cproject files.
+        """
+        opts = {}
+
+        if not self.resources.linker_script:
+            raise NotSupportedException("No linker script found.")
+
+        print ('\nCreate a System Workbench for STM32 managed project')
+        print ('Project name: {0}'.format(self.project_name))
+        print ('Target:       {0}'.format(self.toolchain.target.name))
+        print ('Toolchain:    {0}'.format(self.TOOLCHAIN) + '\n')
```

Comment:
To be honest, I borrowed this from gnuarmeclipse exporter. But I personally like the idea, that program is printing something, so that I know what's going on.

Better option would be to print some deeper information when exporter is called with verbose flag on. Like path to linker script, flags etc. But i suppose there's no way to know if 'mbed export' was called with -v flag?